kopf/reactor/queueing.py

```diff
@@ -247,8 +239,11 @@ def run(
     # The errors in the cancellation stage will be ignored anyway (never re-raised below).
     for task in pending:
         task.cancel()
-    cancelled, pending = loop.run_until_complete(asyncio.wait(pending, return_when=asyncio.ALL_COMPLETED))
-    assert not pending  # must be empty by now, the tasks are either done or cancelled.
+    if pending:
+        cancelled, pending = loop.run_until_complete(asyncio.wait(pending, return_when=asyncio.ALL_COMPLETED))
+        assert not pending  # must be empty by now, the tasks are either done or cancelled.
+    else:
+        cancelled = []
```

Comment:
Can you please clarify what is happening here? 

I know that assertions in the main code are not good. The original idea was that it is impossible that anything is pending at this point. A regular exception can be thrown.

Do I get it right that it is because `asyncio.wait()` on an empty list fails? Still, I didn't get the part with `cancelled = []` — they still have to be `.result()`'ed few lines below.

Reply:
Without this check :243 loop.run_until_complete will fail with
```
ValueError: Set of coroutines/Futures is empty.
```
(no pending tasks left). So this is ugly fix for that case.

Reply:
Ah, got it! If there are no pending tasks, there is nothing to cancel, so there will be no cancelled tasks. But a list is anyway needed few lines below for exception re-raising.